proposals/char-kind-proposal.md

```diff
@@ -0,0 +1,173 @@
+---
+author: Daniel Rogozin, Rinat Stryungis
+date-accepted: ""
+ticket-url: "https://gitlab.haskell.org/ghc/ghc/-/issues/11342"
+implemented: ""
+---
+
+This proposal is [discussed at this merge request](https://gitlab.haskell.org/ghc/ghc/-/merge_requests/3598).
+
+# The Character Kind
+
+We would like to propose and discuss several changes related to the character kind. Some of those changes has been implemented jointly with Rinat Stryungis.
+
+
+## Motivation
+
+The purpose of this patch is to provide a possibility of analysing type-level strings (symbols) as term-level ones. This feature allows users to implement such programs as type-level parsers. One needs to have full-fledged support of type-level characters as well as we already have for strings and numbers. In addition to this functionality, it makes sense to introduce the set of type-families, counterparts of functions defined in the Data.Char module in order to work with type-level strings and chars as usual (more or less).
+
+For more convenience, it’s worth having some of the character-related type families as built-ins and generating the rest of ones as type synonyms.
+
+
+## Proposed Change Specification
+
+The patch fixes #11342, an issue opened by Alexander Vieth several years ago. In this patch, we introduced the Char Kind, a kind of type-level characters, with the additional type-families, type-level counterparts of functions from the `Data.Char` module.
```

Comment:
> The patch fixes #11342, an issue opened by Alexander Vieth several years ago.

Talking about the ticket belongs to the Motivation section. Here, you must describe the actual changes, not their context. Also, make the ticket number into a link to the GitLab issue.